### PR TITLE
feat: tab mockups structural update

### PR DESCRIPTION
**Pull request by mehaknasir**
Supporting Ticket: https://openedx.atlassian.net/browse/TNL-9294
Parent Task: https://openedx.atlassian.net/browse/TNL-8703

Tab Views
Screenshot 2021-11-22 at 8 27 11 PM

Community View
Screenshot 2021-12-03 at 12 32 12 PM

Live View 

is not implemented yet

### CODE REVIEW

**Comment by awaisdar001** [comment]
@mehaknasir can you add screenshot of the mockups implemented?

File: src/components/program/TabularView.jsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  ];
+
+  const [activeTab, setActiveTab] = useState(courseTabs[0].key);
+
+  function changeSelectedTab(tab) {
+    setActiveTab(tab);
+  }
+
+  return (
+    <Tabs activeKey={activeTab.key} id="controlled-tab-example" onSelect={tab => changeSelectedTab(tab)}>
```
Comment (awais-ansari) [inline src/components/program/TabularView.jsx:28]: nit: 
onSelect={tab => changeSelectedTab(tab) => onSelect={tab => setActiveTab(tab).
it approach is better than single liner function.

File: src/components/program/data/reducer.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  FETCH_PROGRAM_DISCUSSIONS_REQUEST,
+  FETCH_PROGRAM_DISCUSSIONS_SUCCESS,
+  FETCH_PROGRAM_DISCUSSIONS_FAILURE,
+} from './constants';
+
+const initialState = {
+  loading: 0,
+  data: null,
+  error: null,
+
```
Comment (awais-ansari) [inline src/components/program/data/reducer.js:11]: nit: extra space

File: src/components/program/tab-components/JourneyTab.jsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+function JourneyTab() {
+  return (
+    <div className="container py-5">
+      <div className="row">
+        <MainContent>
+          <ProgramMainContent />
+        </MainContent>
+        <MediaQuery minWidth={breakpoints.large.minWidth}>
+          {matches => matches && (
```
Comment (awais-ansari) [inline src/components/program/tab-components/JourneyTab.jsx:18]: Please let me know that where these matches are coming from?

Comment (mehaknasir) [reply src/components/program/tab-components/JourneyTab.jsx:18]: https://www.npmjs.com/package/react-responsive

you can see with components example here and see its usage. this matches the property we passed with window size and render components accordingly. matches is the property of MediaQuery and is not declared by us anywhere

File: src/components/program/TabularView.jsx (hunk 1 above)
Comment (mehaknasir) [reply src/components/program/TabularView.jsx:28]: I was doing multiple operations before. on refactoring all those steps are removed, Thanks for pointing out.

File: .env.development [hunk 4]
```diff
@@ -23,3 +23,4 @@ ENABLE_NOTICES=''
 # IDP_SLUG=''
 # uncomment the line below to host the site at /{HOSTNAME}/
 # ENABLE_PATH_PREFIX = true
+USE_MOCK_DATA=''
```
Comment (awaisdar001) [inline .env.development:26]: Why are you using this env variable?

Comment (mehaknasir) [reply .env.development:26]: was mentioned in the portal configuration I see no use of it however but left it there as this env file is just used during dev. mode.

**Review by awaisdar001 — APPROVED** [review]
👍 looks good to me.

**Pull request merged**